src/module/Provider.sol

```diff
@@ -73,4 +77,78 @@ contract Provider is IProvider {
 
         revert UnsupportedAsset(asset);
     }
+
+    /**
+     * @notice function to handle the assets that are in queue for withdrawal.
+     * @param asset_ The address of the asset.
+     * @dev This function should return the amount in base denomination.
+     */
+    function asyncWithdrawalBalance(address asset_) public view returns (uint256 baseAssets) {
+        baseAssets = _asyncWithdrawalBalance(asset_);
+    }
+
+    function _asyncWithdrawalBalanceYNAsset(address queueManager_, address asset_)
+        private
+        view
+        returns (uint256 baseAssets)
+    {
+        IWithdrawalQueueManager queueManager = IWithdrawalQueueManager(queueManager_);
+        (, IWithdrawalQueueManager.WithdrawalRequest[] memory requests) =
+            queueManager.withdrawalRequestsForOwner(msg.sender);
+
+        uint256 decimals = 10 ** ERC20(asset_).decimals();
+
+        for (uint256 i = 0; i < requests.length; i++) {
+            if (!requests[i].processed) {
+                // NOTE: needs to be fixed - assumes no slashing for now, as in reality eigenlayer slashing is not active yet
+                // get base amount
+                uint256 baseAmount = requests[i].amount * requests[i].redemptionRateAtRequestTime / decimals;
+                // get fee
+                uint256 fee = baseAmount * requests[i].feeAtRequestTime / 1000000;
+                // add base amount minus fee
+
+                baseAssets += baseAmount - fee;
+            }
+        }
+        return baseAssets;
+    }
+
+    function _asyncWithdrawalBalanceWSTETH() private view returns (uint256 baseAssets) {
+        IWithdrawalQueue queue = IWithdrawalQueue(MC.WSTETH_WITHDRAWAL_QUEUE);
+        uint256[] memory requestIds = queue.getWithdrawalRequests(msg.sender);
+        IWithdrawalQueue.WithdrawalRequestStatus[] memory statuses = queue.getWithdrawalStatus(requestIds);
+        for (uint256 i = 0; i < statuses.length; i++) {
+            baseAssets += statuses[i].amountOfStETH;
+        }
+    }
+
+    function _asyncWithdrawalBalance(address asset) private view returns (uint256 baseAssets) {
+        if (asset == MC.WOETH) {
+            IOETHVault oethVault = IOETHVault(MC.OETH_VAULT);
+
+            // TODO: see if we should add the withdrawer contracts to Contracts.sol and use it here instead of msg.sender
+            uint256[] memory requestIds = IWithdrawer(msg.sender).getWOETHRequestIds();
```

Comment:
This is probably the only place we need to think about a little bit. It's very specific to the OETH use case.. @danoctavian what's your opinion on this?